gandalf.js

```diff
@@ -1,129 +1,126 @@
 /**
```

Comment:
SCRIPT-This new feature lacks unit tests. Could you add some to ensure everything works correctly?